### PR TITLE
feat: add baseline streaming for agents

### PR DESCRIPTION
**Pull request by olbychos**
add baseline streaming support for react agent with ability to stream intermediate steps/final answer only, for simple/reflection agent

### CODE REVIEW

File: dynamiq/nodes/agents/base.py [hunk 1]
```diff
@@ -57,7 +57,9 @@ class Agent(Node):
     llm: Node = Field(..., description="LLM used by the agent.")
     group: NodeGroup = NodeGroup.AGENTS
     error_handling: ErrorHandling = ErrorHandling(timeout_seconds=600)
-    streaming: StreamingConfig = StreamingConfig()
+    streaming_mode: StreamingMode = Field(
```
Comment (delamainer) [inline dynamiq/nodes/agents/base.py]: I would keep it as `streaming`. Essentially, this is a still node's steaming configuration, but with a different set of parameters.

Comment (olbychos) [reply dynamiq/nodes/agents/base.py]: Already updated this!

Comment (delamainer) [reply dynamiq/nodes/agents/base.py]: Do we use the same event name for the final output and intermediate steps?

**Review by TrachukT — APPROVED** [review]

File: dynamiq/types/streaming.py [hunk 2]
```diff
@@ -65,12 +65,14 @@ class StreamingConfig(BaseModel):
         timeout (float | None): Timeout for streaming. Defaults to None.
         input_queue (Queue | None): Input queue for streaming. Defaults to None.
         input_queue_done_event (Event | None): Event to signal input queue completion. Defaults to None.
+        is_final (bool): Whether the streaming is final. Defaults to False.
```
Comment (acoola) [inline dynamiq/types/streaming.py:68]: Sorry, both the name of param and description are not clear.

Comment (olbychos) [reply dynamiq/types/streaming.py:68]: Option for stream all steps or only final version

Comment (acoola) [reply dynamiq/types/streaming.py:68]: Let's use mode with Enum as you used previously

File: examples/use_case_streaming_agents/react/backend.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    return agent
+
+
+def generate_agent_response(agent: ReActAgent, user_input: str):
+    """
+    Processes the user input using the agent. Supports both streaming and non-streaming responses.
+    """
+    if agent.streaming.enabled:
+        streaming_handler = StreamingIteratorCallbackHandler()
+        agent.run(
```
Comment (acoola) [inline examples/use_case_streaming_agents/react/backend.py:42]: To unblock this operation, don't wait result of it and start streaming immediately you could run it in separate thhread. Im similar way we managed streaming in async:
https://github.com/dynamiq-ai/dynamiq/blob/main/examples/chainlit/chainlit_agent.py#L76
https://github.com/dynamiq-ai/dynamiq/blob/main/examples/chainlit/chainlit_agent.py#L26

**Pull request merged**